masto-module/src/main/kotlin/au/rakka/java/masto-api/MastoReplyGuy.kt

```diff
@@ -0,0 +1,92 @@
+	
+	private val baseurl = "https://${conf.instance}/api/v1/" // Well it was half useful
+	private val notifurl = "${baseurl}notifications"
+	private val clearurl = "${baseurl}notifications/clear"
+	private val statusurl = "${baseurl}statuses"
+	
+	override val schedule: Int = 30
+	override suspend fun task()
+	{
+		logger.debug("Blep")
+		var finished=false
+		var max_id=""
+		while (!finished)
+		{
+			val response = client.sendAsync(builder.uri(URI.create(notifurl+max_id)).GET().build(),HttpResponse.BodyHandlers.ofInputStream()).await()
+			logger.debug(response.statusCode().toString())
+			val node=mapper.readTree(response.body())
+			node.forEach{ process_post(it) }
+			logger.debug(node.size().toString())
+			finished=node.size()<20 // If node size isn't 20 I don't need to update max_id at all. Mainly avoids the issue where node size is 0.
+			if (!finished) {max_id=node.get(19).get("id").asText()}
+		}
+		val response = client.sendAsync(builder.uri(URI.create(clearurl)).POST(HttpRequest.BodyPublishers.ofString("")).build(),HttpResponse.BodyHandlers.ofInputStream()).await()
+		logger.debug("Cleared notifications (hopefully) "+response.statusCode().toString())
+	}
+	private suspend fun process_post(post:JsonNode)
+	{
+		logger.debug("Processing notif "+post.get("id"))
+		if (post.get("type").asText()!="mention") {logger.debug("Was not a mention");return}
+		val text= post.get("status").get("pleroma").get("content").get("text/plain").asText()
+		val (tag,regex)=process_post_contents(text)
+		if (tag=="") {logger.debug("tag: ${tag} was null");return}
+		if (regex=="")
+		{
+			logger.debug("Deleting ${tag}")
+			wc3GameNotificationService.deleteNotification(tag) // lol hope you meant it
+			val response=client.sendAsync(builder.uri(URI.create(statusurl)).POST(HttpRequest.BodyPublishers.ofString("""{"status":"Unregistered ${tag}. In future this will hopefully be able to tell you what it contained.","in_reply_to_id":${post.get("status").get("id")}}""")).build(),HttpResponse.BodyHandlers.ofInputStream()).await()
+			logger.debug("Deleted "+response.statusCode().toString()+" "+post.get("status").get("id"))
+		}
+		else
+		{
+			logger.debug("Adding ${tag} with ${regex}")
+			wc3GameNotificationService.createNotification(tag,regex)
+			val response=client.sendAsync(builder.uri(URI.create(statusurl)).POST(HttpRequest.BodyPublishers.ofString("""{"status":"Registered ${tag} with pattern ${regex}.","in_reply_to_id":${post.get("status").get("id")}}""")).build(),HttpResponse.BodyHandlers.ofInputStream()).await() // .get("id") returns a string with quotes: Deleted 400 "AotNh4gQIua2IPvPN2", so don't need to put new quotes on it.
+			logger.debug("Created "+response.statusCode().toString()+" "+post.get("status").get("id"))
+		}
+	}
+	private fun process_post_contents(contents:String) : Pair<String,String>
+	{
+		logger.debug("Processing post contents")
+		var tag=""
+		var regex=""
+		val words=contents.split(' ')
+		for (word in words)
```

Comment:
You can actually do `words.forEach { word -> }` if you prefer.

Reply:
Is it guaranteed to execute in order?
And in general, I prefer the flow of quickly exiting when I know there's nothing to do, which is non-trivial in those.

Reply:
As far as I know it's guaranteed to execute in order, although I've never actually tried it. Exiting early is a valid concern, but Kotlin actually supports that with inline functions and non-local returns in lambdas. If you determine that you don't need to keep looping, I think you can do `return@forEach` to break out of the iteration... maybe. You could do `return@process_post_contents` to break out of the entire function.

Reply:
I rate it as "more effort than it's worth", given that I could receive 10 tags in a row before a regex, or 10 regexes in a row before a tag.
Though to be honest I don't really care what the behaviour is in either of those scenarios, just don't do stupid things™.

Reply:
So you can see that `forEach` actually literally does what you've got here. 😛 https://github.com/JetBrains/kotlin/blob/master/libraries/stdlib/common/src/generated/_Collections.kt#L1862-L1864